Move schema to individual package

0b5vr opened the pull request:
### ~~Why DO NOT MERGE~~

- ~~The breaking change is too big, we should wait till VRM1.0 support~~

### About

- This PR provides schema of VRM as individual packages
- ~~🚨: We are going to no longer be not able to use `VRMSchema.HumanoidBoneName` and `VRMSchema.BlendShapePresetName` as enums~~
    - ~~Since we're going to make them union types instead of enums~~
    - ~~We should provide a deprecation warning and compat for a while though, it's too terrible to disable it immediately~~
    - **Update:** union type + const value should be included in individual implementation modules
    - **Update:** Now it should not have any changes other than adding schema packages

Commit:
refactor: make VRM 0.0 schema an individual package

Commit:
refactor: make VRM 1.0 schema an individual package (as of 0a10fcdd)

Commit:
fix (types-vrm-0.0, root): materialProperties is array

Commit:
refactor (types-vrm-0.0): BlendShapePresetName and HumanoidBoneName is now union types instead of enums

Commit:
feat: add a package, types-vrmc-constraints-1.0

Commit:
chore (CI): Add cache directories

+ packages/types-vrm-0.0/node_modules
+ packages/types-vrmc-vrm-1.0/node_modules

Commit:
fix (types-vrmc-constraints): add forgotten index exports

Commit:
docs (types-vrmc-constraints): add a ref on readme

Commit:
change (types, vrm1.0): Apply changes of the expression PR
Ref: https://github.com/vrm-c/vrm-specification/pull/171

Commit:
change (types, vrm1.0): Apply changes of the expression schema updates PR
Ref: https://github.com/vrm-c/vrm-specification/pull/187

Commit:
change (types, vrm1.0): Apply changes of the meta licenses PR
Ref: https://github.com/vrm-c/vrm-specification/pull/184

Commit:
change (constraint): change casing of object space enums

Commit:
change (types, constraint): Space -> ConstraintsObjectSpace

Commit:
deps: remove devDeps for each package that are no longer required

Commit:
feature: add a new package, types-vrmc-materials-mtoon-1.0

Commit:
refactor: fix a lint issue

Commit:
chore: update tsconfig.json

Commit:
fix (types, material): forgot to export union types from index

Commit:
chore (CI): add a cache path

Commit:
chore (CI): modify use of yarn

Commit:
feature: init types-node-collider-1.0

Commit:
feature: init types-node-springbone-1.0

Commit:
chore (CI): add a cache directory
+ packages/types-vrmc-constraints-1.0/node_modules

Commit:
chore: update tsconfig.json

FMS-Cat commented:
Changed my mind: `VRMSchema.HumanoidBoneName` should be exist as enums since Three.js uses such kind of enums

Commit:
refactor: add enums
+ VRMBlendShapePresetName
+ VRMHumanoidBoneName

Commit:
refactor: move enums to even more appropriate place

Commit:
refactor: replace local VRMSchema with @pixiv/types-vrm-0.0

Commit:
Merge branch 'rollup-2' into schema-package

Commit:
chore: update docs commands

Commit:
change (types-vrmc-vrm-1.0): apply override expressions changes
The PR: https://github.com/vrm-c/vrm-specification/pull/196
I believe it's now even with: https://github.com/vrm-c/vrm-specification/commit/512a3b4303b54430ee34286fbbe4d8c9234d8f54

Commit:
Merge branch 'dev' into schema-package

Commit:
change (types-springbone-1.0): apply latest changes of schema
corresponds to https://github.com/vrm-c/vrm-specification/pull/212

Commit:
fix (types-springbone-1.0): add missing glTFProperty props

Commit:
change (types-vrmc-vrm-1.0): apply latest changes of schema
corresponds to https://github.com/vrm-c/vrm-specification/pull/221/

Commit:
refactor (types-vrm, expressions): change order of expression presets
corresponds to https://github.com/vrm-c/vrm-specification/pull/218

Commit:
fix (types, firstperson): add glTFProperty fields to firstperson

Commit:
fix (types): several property updates
- FirstPersonMeshAnnotation, node and type are now required
- FirstPersonMeshAnnotation, firstPersonType -> type
- LookAt, lookAtType -> type
- LookAt, lookAt**** -> rangeMap****
corresponds to https://github.com/vrm-c/vrm-specification/pull/239

Commit:
fix (types): add glTFProperty fields to the root of vrm

Commit:
fix (types, meta): add glTFProperty fields to meta

Commit:
fix (types, humanoid): add glTFProperty fields to humanoid properties

Commit:
fix (types, lookat): add glTFProperty fields to lookat properties

Commit:
change (types, meta): update schema of meta
corresponds to https://github.com/vrm-c/vrm-specification/pull/253

Commit:
change (types, springbone) move node collider types to springbone

Commit:
change (types, springbone): rename node collider to spring bone collider (1)

Commit:
change (types, springbone): rename node collider to spring bone collider (2)

Commit:
change (types, springbone): update springbone schema

Commit:
change (types, springbone): update mtoon schema

Commit:
fix (types, springbone): expose SpringBoneColliderGroup

Commit:
Merge branch 'dev' into schema-package

Commit:
fix (types, springbone): shapes -> shape

Commit:
Merge branch 'dev' into schema-package

Commit:
change (1.0, schema): add licenseUrl to meta

Commit:
docs (1.0, schema): update doc comments of root

Commit:
change (1.0, schema): update schema of expressions

Commit:
docs (1.0, schema): update the doc comment of LookAtRangeMap.outputScale

Commit:
fix (1.0, schema): remove custom from expression preset

Commit:
fix (1.0, schema): make expression properties optional

Commit:
change (1.0, schema): update type definition of VRM root

Commit:
change (1.0, schema): add specVersion for constraints

Commit:
change (1.0, schema): add specVersion for springBone

Commit:
change (1.0, schema): mtoon, change version to specVersion

Commit:
change (1.0, schema): ExpressionPreset -> ExpressionPresetName

Commit:
refactor (1.0, schema): tidy up root index.ts

Commit:
docs (1.0, schema): update readme of types-vrm-0.0

Commit:
change (1.0, schema): VRM -> VRMCVRM

Commit:
change (1.0, schema): types-vrmc-constraints-1.0 -> types-vrmc-node-constraints-1.0

Commit:
change (1.0, schema): Constraints -> VRMCNodeConstraint
ConstraintsObjectSpace -> ConstraintObjectSpace

Commit:
change (1.0, schema): SpringBone -> VRMCSpringBone

Commit:
change (1.0, schema): MToonMaterial -> VRMCMaterialsMToon

Commit:
feature (1.0, schema): add a package, types/vrmc-materials-hdr-emissive-multiplier-1.0

Commit:
chore (1.0): on `schema-package` branch, rewind three-vrm back to dev state

0b5vr commented:
rewinded all the changes about `three-vrm`, now it should not have any changes other than adding schema packages
